src/lib/one-graph/client.js

```diff
@@ -0,0 +1,835 @@
+    noLocation: true,
+  })
+  const operations = extractFunctionsFromOperationDoc(parsedDoc)
+  generateFunctionsFile(netligraphConfig, schema, appOperationsDoc, operations)
+}
+
+const friendlyEventName = (event) => {
+  const { __typename, payload } = event
+  switch (__typename) {
+    case 'OneGraphNetlifyCliSessionTestEvent':
+      return friendlyEventName(payload)
+    case 'OneGraphNetlifyCliSessionGenerateHandlerEvent':
+      return 'Generate handler as Netlify function '
+    case 'OneGraphNetlifyCliSessionPersistedLibraryUpdatedEvent':
+      return `Sync Netlify Graph operations library`
+    default: {
+      return `Unrecognized event (${__typename})`
+    }
+  }
+}
+
+const handleCliSessionEvent = async ({ authToken, event, netligraphConfig, schema, siteId }) => {
+  const { __typename, payload } = await event
+  switch (__typename) {
+    case 'OneGraphNetlifyCliSessionTestEvent':
+      await handleCliSessionEvent({ authToken, event: payload, netligraphConfig, schema, siteId })
+      break
+    case 'OneGraphNetlifyCliSessionGenerateHandlerEvent':
+      await generateHandler(netligraphConfig, schema, payload.operationId, payload)
+      break
+    case 'OneGraphNetlifyCliSessionPersistedLibraryUpdatedEvent':
+      await updateGraphQLOperationsFile({ authToken, docId: payload.docId, netligraphConfig, schema, siteId })
+      break
+    default: {
+      console.warn(
+        `${NETLIFYDEVWARN} Unrecognized event received, you may need to upgrade your CLI version`,
+        __typename,
+        payload,
+      )
+      break
+    }
+  }
+}
+
+const loadCLISession = (state) => state.get('oneGraphSessionId')
+
+const startOneGraphCLISession = async ({ netlifyToken, netligraphConfig, site, state }) => {
+  let oneGraphSessionId = loadCLISession(state)
+  if (!oneGraphSessionId) {
+    const sessionName = generateSessionName()
+    const oneGraphSession = await createCLISession(netlifyToken, site.id, sessionName)
+    state.set('oneGraphSessionId', oneGraphSession.id)
+    oneGraphSessionId = state.get('oneGraphSessionId')
+  }
+
+  const enabledServices = []
+  const schema = await fetchOneGraphSchema(site.id, enabledServices)
+
+  monitorCLISessionEvents({
+    appId: site.id,
+    authToken: netlifyToken,
+    netligraphConfig,
+    sessionId: oneGraphSessionId,
+    state,
+    onEvents: (events) => {
+      events.forEach(async (event) => {
+        const eventName = friendlyEventName(event)
+        log(`${NETLIFYDEVLOG} ${chalk.magenta('Handling')} Netlify Graph event: ${eventName}...`)
+        await handleCliSessionEvent({ authToken: netlifyToken, event, netligraphConfig, schema, siteId: site.id })
+        log(`${NETLIFYDEVLOG} ${chalk.green('Finished handling')} Netlify Graph event: ${eventName}...`)
+      })
+      return events.map((event) => event.id)
+    },
+    onError: (error) => {
+      console.error(`OneGraph: ${error}`)
+    },
+    onClose: () => {
+      console.debug('OneGraph: closed')
```

Comment:
maybe we want to leverage the debug package with debug scopes for that:

```javascript
const debug = require('debug')

debug(`onegraph`)('closed')
```

This will be then shown only if the command was run with `DEBUG=* ntl ...` or `DEBUG=onegraph ntl ...`